test/rulesets/Base/indexing.jl

```diff
@@ -1,3 +1,17 @@
+@testset "getfield" begin
+    struct Foo
+        x::Float64
+        y::Float64
+    end
+    test_frule(getfield, Foo(1.5, 2.5), :x, check_inferred=false)
+    
+    test_frule(getfield, (; a=1.5, b=2.5), :a, check_inferred=false)
```

Comment:
These actually infer fine as long as the `frule` gets inlined somewhere where the symbol is known at compile time.
I.e. almost all uses hopefully.